homeassistant/components/sensor/citybikes.py

```diff
@@ -0,0 +1,186 @@
+                break
+            yield station
+    else:
+        for station in network.stations:
+            if station['id'] in stations_list:
+                yield station
+                continue
+            if 'extra' in station:
+                if 'uid' in station['extra']:
+                    if str(station['extra']['uid']) in stations_list:
+                        yield station
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the CityBikes platform."""
+    from citybikes import Client, Network
+
+    latitude = config.get(CONF_LATITUDE, hass.config.latitude)
+    longitude = config.get(CONF_LONGITUDE, hass.config.longitude)
+
+    network_uid = config.get(CONF_NETWORK)
+    if not network_uid:
+        network = Client().networks.near(latitude, longitude)[0][0]
+    else:
+        network = Network(Client(), uid=network_uid)
+
+    stations_list = config.get(CONF_STATIONS_LIST, [])
+    radius = config.get(CONF_RADIUS, 0)
+    if not hass.config.units.is_metric:
+        radius = distance.convert(radius, LENGTH_FEET, LENGTH_METERS)
+    poller = CityBikesNetworkPoller(network)
+
+    add_devices(CityBikesStationSensor(poller, station,
+                                       config.get(CONF_NAME, DOMAIN))
+                for station in _filter_stations(network, radius, latitude,
+                                                longitude, stations_list))
+
+
+class CityBikesNetworkPoller(object):
+    """A wrapper around a CityBikes Network object."""
+
+    def __init__(self, network):
+        """Initialize the poller."""
+        self._network = network
+
+    def get(self, station_id, update=True):
+        """Return the station with the given id."""
+        if update:
+            self._update()
+        for station in self._network.stations:
+            if station[ATTR_STATION_ID] == station_id:
+                return station
+
+    @Throttle(SCAN_INTERVAL)
+    def _update(self):
+        """Update the state of the network."""
+        self._network.request()
+
+
+class CityBikesStationSensor(Entity):
+    """CityBikes API Sensor."""
+
+    def __init__(self, poller, station, base_name):
+        """Initialize the sensor."""
+        self._poller = poller
+        self._base_name = base_name
+        self._update(station)
+
+    def _update(self, station):
+        self._id = station[ATTR_STATION_ID]
+        self._uid = station[ATTR_EXTRA]['uid'] if ATTR_EXTRA in station \
+            and 'uid' in station[ATTR_EXTRA] \
+            else None
+        self._latitude = station[ATTR_LATITUDE]
+        self._longitude = station[ATTR_LONGITUDE]
+        self._empty_slots = station[ATTR_EMPTY_SLOTS]
+        self._free_bikes = station[ATTR_FREE_BIKES]
+        self._timestamp = station[ATTR_TIMESTAMP]
+        self._friendly_name = station[ATTR_STATION_NAME]
+        self._name = slugify("{}_{}".format(self._base_name,
```

Comment:
Names don't need to be slugified, instead they should be a real name including spaces.